ports/bare-arm/lib.c

```diff
@@ -110,8 +110,9 @@ char *strchr(const char *s, int c) {
 }
 
 int strncmp(const char *s1, const char *s2, size_t n) {
-    while (*s1 && *s2 && n-- > 0) {
+    while (n > 0 && *s1 && *s2) {
```

Comment:
Ha, this actually fixes the case when `n=0` is passed in to this function, with both strings non-null.